docs/book/v5/tutorials/find-user-by-identity.md

```diff
@@ -0,0 +1,213 @@
+
+```php
+$app->get('/user/' . $uuid, UserHandler::class, 'user.view');
+```
+
+Our route points to `get` method from `UserHandler` so let's navigate to that method.
+
+```php
+public function get(ServerRequestInterface $request): ResponseInterface
+{
+    $user = $this->userService->findOneBy(['uuid' => $request->getAttribute('uuid')]);
+
+    return $this->createResponse($request, $user);
+}
+```
+
+As we can see, the method will query the database for the user based on its uuid taken from the endpoint.
+
+We now have an understanding of how things work and we can start to implement our own endpoint.
+
+### Implementation
+
+We need to create a new handler that will process our request, we can call it `IdentityHandler`.
+
+Create a new PHP class called `IdentityHandler.php` in `src/User/src/Handler` folder.
+
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace Api\User\Handler;
+
+use Api\App\Exception\BadRequestException;
+use Api\App\Exception\NotFoundException;
+use Api\App\Handler\HandlerTrait;
+use Api\App\Message;
+use Api\User\Entity\User;
+use Api\User\Service\UserServiceInterface;
+use Dot\DependencyInjection\Attribute\Inject;
+use Mezzio\Hal\HalResponseFactory;
+use Mezzio\Hal\ResourceGenerator;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+use function sprintf;
+
+class IdentityHandler implements RequestHandlerInterface
+{
+    use HandlerTrait;
+
+    #[Inject(
+        HalResponseFactory::class,
+        ResourceGenerator::class,
+        UserServiceInterface::class,
+    )]
+    public function __construct(
+        protected HalResponseFactory $responseFactory,
+        protected ResourceGenerator $resourceGenerator,
+        protected UserServiceInterface $userService,
+    ) {
+    }
+
+    /**
+     * @throws NotFoundException
+     * @throws BadRequestException
+     */
+    public function get(ServerRequestInterface $request): ResponseInterface
+    {
+        $identity = $request->getAttribute('identity');
+        if (empty($identity)) {
+            throw (new BadRequestException())->setMessages([sprintf(Message::INVALID_VALUE, 'identity')]);
+        }
+
+        $user = $this->userService->findByIdentity($identity);
+        if (! $user instanceof User) {
+            throw new NotFoundException(Message::USER_NOT_FOUND);
+        }
+
+        return $this->createResponse($request, $user);
+    }
+}
+```
+
+Our handler is very similar to the existing one, with some extra steps:
+
+* We store the identity from the request in the `$identity` variable for later use.
+* If the identity is empty we throw a `BadRequestException` with an appropriate message.
+* If we can't find the user in the database we throw an `NotFoundException`.
+* We generate and return the response.
+
+The next step is to register the new handler,

```

Comment:
update for punctuation, for clarity
`The next step is to register the new handler,`
to
`The next step is to register the new handler.`